lib/HLSL/HLOperationLower.cpp

```diff
@@ -3921,9 +3921,16 @@ void TranslateStore(DxilResource::Kind RK, Value *handle, Value *val,
 
     // For second and subsequent store calls, increment the offset0 (i.e. store index)
     if (j > 0) {
-      Value* newOffset = ConstantInt::get(Builder.getInt32Ty(), j);
-      newOffset = Builder.CreateAdd(storeArgsList[0][offset0Idx], newOffset);
-      storeArgsList[j][offset0Idx] = newOffset;
+      // Greater than four-components store is not allowed for
+      // TypedBuffer and Textures. Also, StructuredBuffer store
+      // doesn't use this function. So greater than four elements
```

Comment:
If we are sure that:

> Also, StructuredBuffer store doesn't use this function.

We should put a DXASSERT at the top of function and remove the case in the switch that looks like it handles StructuredBuffer.

Reply:
Added an assert at the function entry.